packages/amplication-client/src/Project/ArchitectureConsole/hooks/useModelOrganizer.tsx

```diff
@@ -553,11 +555,32 @@ const useModelOrganization = ({ projectId, onMessage }: Props) => {
         (node) => node.id === sourceParentNodeId
       ).data.payload.name;
 
+      const currentTargetResource: ResourceNode = targetParent as ResourceNode;
+
       movedNodes.forEach((node) => {
-        const currentNode = currentNodes.find((n) => n.id === node.id);
+        const currentNode: EntityNode = currentNodes.find(
+          (n) => n.id === node.id
+        ) as EntityNode;
+
+        const duplicatedEntityName =
+          currentTargetResource.data.payload.entities.find(
```

Comment:
you only check the entities in the target resource, but you also need to check the recent moved entities (from the changes)

Reply:
we only alloe moves from a single source, so this cannot happen